finagle-redis/src/main/scala/com/twitter/finagle/redis/protocol/commands/PubSub.scala

```diff
@@ -0,0 +1,96 @@
+package com.twitter.finagle.redis.protocol
+
+import org.jboss.netty.buffer.ChannelBuffer
+import org.jboss.netty.buffer.ChannelBuffers
+import com.twitter.finagle.redis.ClientError
+import com.twitter.finagle.redis.protocol.Commands._
+import com.twitter.finagle.redis.util._
+
+abstract class SubscribeCommand extends RedisMessage {
+  def command: String
+  def listener: SubscribeListener
+}
+
+trait SubscribeListener {
+  def onException(ex: Throwable): Unit
+  def onMessage(msg: Reply): Unit
+}
+
+case class PSubscribe(patterns: Seq[ChannelBuffer], listener: SubscribeListener) extends SubscribeCommand {
```

Comment:
hmmm, would you be OK with doing it at construction-time?  or do you think it's important to be able to do it request-time?